src/main/java/com/based/basedsurvey/controller/FillSurveyController.java

```diff
@@ -0,0 +1,159 @@
+package com.based.basedsurvey.controller;
+import com.based.basedsurvey.data.*;
+import com.based.basedsurvey.repo.SurveyRepository;
+import lombok.extern.java.Log;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import java.util.List;
+
+@Log
+@Controller
+public class FillSurveyController {
+
+    private SurveyRepository surveyRepository;
+
+    @Autowired
+    public FillSurveyController(SurveyRepository surveyRepository) {
+        this.surveyRepository = surveyRepository;
+    }
+
+    @GetMapping("/survey/{surveyID}/answer")
+    public String getAnswer(@PathVariable Long surveyID, Model model) {
+        // if survey does not exist
+        if (!surveyRepository.existsById(surveyID)) {
+            model.addAttribute("issue", "Survey does not exist");
+            return "SurveyFillIssue";
+        }
+
+        model.addAttribute("surveyID", surveyID);
+        Survey s = this.surveyRepository.findSurveyById(surveyID);
+        model.addAttribute("surveyName", s.getName());
+
+        // if survey is closed
+        if (!s.isOpen()) {
+            model.addAttribute("issue", "Survey is closed");
+            return "SurveyFillIssue";
+        }
+
+        // if survey has no questions
+        if (s.getQuestions().size() == 0) {
```

Comment:
Will issue be overwritten if it's closed and size is 0 and if so we may need to account for that in the future